src/core/lib/event_engine/posix_engine/grpc_polled_fd_posix.h

```diff
@@ -93,16 +103,95 @@ class GrpcPolledFdFactoryPosix : public GrpcPolledFdFactory {
   explicit GrpcPolledFdFactoryPosix(PosixEventPoller* poller)
       : poller_(poller) {}
 
+  ~GrpcPolledFdFactoryPosix() override {
+    for (auto& fd : owned_fds_) {
+      close(fd);
+    }
+  }
+
   GrpcPolledFd* NewGrpcPolledFdLocked(ares_socket_t as) override {
+    GPR_ASSERT(owned_fds_.insert(as).second);
```

Comment:
Interesting NOTE: asserting `GPR_ASSERT(owned_fds_.insert(as).second);` here is not correct for EE's `AresResolver`. EE's `AresResolver` reuses the same channel for all 3 types of requests, so it will very likely use the same socket for different requests. If the previous `GrpcPolledFdPosix` has been destroyed, it would create a new one, but `GrpcPolledFdFactoryPosix` has already seen the socket.

Reply:
That is interesting. So if I understand we will create a `GrpcPolledFdPosix` potentially a twice over the same socket?

If so do we have a guarantee that the first `GrpcPolledFdPosix` will not mutate the socket in such a way that it becomes not reusable? E.g. do we have a guarantee that `OrphanHandle` and anything else will not call `shutdown` ?

Reply:
Yeah, we call `OrphanHandle` with a valid `release_fd` pointer and that will prevent the socket from being `shutdown` or `close.`